Investigate edit profile failures

Multiple failures on Affiliates on staging, no reported changes have been introduced.

https://webqa-ci.mozilla.com/job/affiliates.stage/107/HTML_Report/
https://webqa-ci.mozilla.com/job/affiliates.stage/121/HTML_Report/

It seems to focus on this error:
E       WebDriverException: Message: Error forwarding the new session cannot find : Capabilities [{firefox_profile=UEsDBBQAAAAIAD19f0Yo4cn2VAMAA..., platform=MAC, javascriptEnabled=true, browserName=firefox, version=33}]

--- a/pages/user.py
+++ b/pages/user.py
@@ -24,7 +24,6 @@ class EditProfile(Base):
     _milestones_section_locator = (By.ID, 'milestones')
     _newsletter_form_locator = (By.ID, 'newsletter-form')
 
-
     def click_edit_profile(self):
         self.selenium.find_element(*self._edit_profile_locator).click()
         return self.EditProfileModal(self.testsetup)
@@ -139,4 +138,4 @@ class EditProfile(Base):
         def click_cancel(self):
             self._root_element.find_element(*self._cancel_locator).click()
             WebDriverWait(self.selenium, self.timeout).until(
-                lambda s: self.is_element_not_visible(*self._modal_locator))
+                lambda s: not self.is_element_present(*self._modal_locator))

